To be merged for 7.1 - MTA-3848: MTA flags are missing in the documentation (#69)
migtools/mta-documentation: docs/topics/mta-cli-args.adoc

a|`--list-targets`
|
|Flag to list rules for available migration targets.

a|`--maven-settings`
|string
|Flag path to a custom Maven settings file to use

a|`-m`, `--mode`
|String
a|Flag for analysis mode, this must be one of `full`, for `source` and `dependencies`, or `source-only` (default `full`).

a|`--no-proxy`
|String
|Flag to excluded URLs from passing through any proxy (relevant only with proxy)

a|`-o`, `--output`
|String
|Flag for the path to the directory for analysis output. For more details, see xref:cli-output-argument_cli-guide[Specifying the output directory].

a|`--override-provider-settings`
|String
|Flag to override the provider settings. The analysis pod runs on the host network, and no providers are started.

a|`--overwrite`
|
|Flag to overwrite the output directory. If you do not specify this argument and the `--output` directory exists, you are prompted to choose whether to overwrite the contents.

a|`--provider`
|String Array
|Flag to specify which provider or providers to run.

a|`--rules`
|String Array
a|Flag to specify the filename or directory containing rule files. Use multiple times for additional rules, for example, `--rules --rules …`.

> a|`--run-local`

Reviewer: Just note, this is default `true` in 7.2, can be disabled with `--run-local=false`.

Author: Damn it! Will have to do a second PR for that

a|`--jaeger-endpoint`
|String
|Flag for the jaeger endpoint to collect traces.

a|`--json-output`
|
a|Flag to create analysis and dependency output as JSON.

a|`--list-sources`
|
a|Flag to list rules for available migration sources.

a|`--list-targets`
|
|Flag to list rules for available migration targets.

a|`--list-providers`
|
|Flag to list available supported providers.

a|`-l`, `--label-selector`
|String
|Flag to run rules based on a specified label selector expression.

a|`--maven-settings`
|string
|Flag path to a custom Maven settings file to use

a|`--overwrite`
|
|Flag to overwrite the output directory. If you do not specify this argument and the `--output` directory exists, you are prompted to choose whether to overwrite the contents.

a|`--skip-static-report`
|
|Flag to not generate static report.

a|`-m`, `--mode`
|String
a|Flag for analysis mode, this must be one of `full`, for `source` and `dependencies`, or `source-only` (default `full`)

a|`-o`, `--output`
|String
|Flag for the path to the directory for analysis output. For more details, see xref:cli-output-argument_cli-guide[Specifying the output directory].

a|`--override-provider-settings`
|String

> |Flag to override the provider settings, the analysis pod runs on the host network, and no providers are started up.

Reviewer 2: ```suggestion
|Flag to override the provider settings. The analysis pod runs on the host network, and no providers are started.
```

a|`--provider`
|String Array
|Flag to specify which provider or providers to run.

a|`--rules`
|String Array

> a|Flag to specify the filename or directory containing rule files. Use multiple times for additional rules: `--rules --rules …`

Reviewer 2: ```suggestion
a|Flag to specify the filename or directory containing rule files. Use multiple times for additional rules, for example, `--rules --rules …`.
```

a|`-s`, `--source`
|String Array

> a|Flag for the source technology to consider for analysis. Use multiple times for additional sources: `--source --source …`. For more details, see xref:cli-source-argument_cli-guide[Setting the source technology].

Reviewer 2: ```suggestion
a|Flag for the source technology to consider for analysis. Use multiple times for additional sources, for example, `--source --source …`. For more details, see xref:cli-source-argument_cli-guide[Setting the source technology].
```

a|`-t`, `--target`
|String Array

> a|Flag for the target technology to consider for analysis. Use multiple times for additional targets: `--target --target …`. For more details, see xref:cli-target-argument_cli-guide[Setting the target technology].

Reviewer 2: ```suggestion
a|Flag for the target technology to consider for analysis. Use multiple times for additional targets, for example, `--target --target …`. For more details, see xref:cli-target-argument_cli-guide[Setting the target technology].
```

//
// * docs/cli-guide/master.adoc

:_content-type: REFERENCE
[id="cli-args_{context}"]
= About {ProductShortName} command-line arguments

The following is a detailed description of the available {ProductShortName} command line arguments.

[NOTE]
====
To run the {ProductShortName} command without prompting, for example when executing from a script, you must use the following arguments:

> * `--batchMode`

Reviewer: There in no --batchMode in current CLI.

* `--batchMode`
* `--overwrite`
* `--input`
* `--target`
====

.{ProductShortName} CLI arguments
[cols="40%,60%a",options="header",]
|====
|Argument |Description
|--analyze-known-libraries | Flag to analyze known software artifacts embedded within your application. By default, {ProductShortName} only analyzes application code.

[NOTE]
[%collapsible]
====
This option might result in a longer execution time and a large number of migration issues being reported.
====

|--help |Display the {ProductShortName} help message.
|--input |A space-delimited list of the path to the file or directory containing one or more applications to be analyzed. This argument is required.
| --list-sources
| list rules for available migration sources
|--list-sources | Flag to list all available source technologies.
|--list-targets| Flag to list all available target technologies.

> |--output |Specify the path to the directory to output the report information generated by {ProductShortName}.

Reviewer: All flags in section are still present in CLI.

Author: Thanks, I have included them in the table

Please can I ask for clarification on `--list-sources` flag as there are two definitions in the above. Is it correct to say:

| | |
| --- | --- | --- |
|`--list-sources` | |Flag to list rules for available migration sources. |

a|`--skip-static-report`
|
|Flag to not generate static report.

a|`-s`, `--source`
|String Array
a|Flag for the source technology to consider for analysis. Use multiple times for additional sources, for example, `--source --source …`. For more details, see xref:cli-source-argument_cli-guide[Setting the source technology].

a|`-t`, `--target`
|String Array
a|Flag for the target technology to consider for analysis. Use multiple times for additional targets, for example, `--target --target …`. For more details, see xref:cli-target-argument_cli-guide[Setting the target technology].
|====
====

[id="cli-input-argument_{context}"]
== Specifying the input

A space-delimited list of the path to the file or directory containing one or more applications to be analyzed. This argument is required.

.Usage
[source,options="nowrap",subs="attributes+"]
----
--input <INPUT_ARCHIVE_OR_DIRECTORY> [...]
----

Depending on whether the input file type provided to the `--input` argument is a file or directory, it will be evaluated as follows depending on the additional arguments provided.

Directory::
+
[cols="1,1,1",options="header"]
|====

> | --explodedApp

Reviewer: There is no explodedApp or sourceMode anymore in CLI.

//
// * docs/cli-guide/master.adoc

:_content-type: REFERENCE
[id="cli-args_{context}"]
= About {ProductShortName} command-line arguments

The following is a detailed description of the available {ProductShortName} command line arguments.

[NOTE]
====
To run the {ProductShortName} command without prompting, for example when executing from a script, you must use the following arguments:

* `--batchMode`
* `--overwrite`
* `--input`
* `--target`
====

.{ProductShortName} CLI arguments
[%collapsible]
====

> [cols="40%,60%a",options="header",]

Author: ```suggestion
```

--output <OUTPUT_REPORT_DIRECTORY>
----

* If omitted, the report will be generated in an `<INPUT_ARCHIVE_OR_DIRECTORY>.report` directory.

> * If the output directory exists, you will be prompted with the following (with a default of N).

Reviewer 2: ```suggestion
* If the output directory exists, you will be prompted with the following question with a default answer of `N`:
```
Is it really required to state the the N answer is the default one? I'd assume user should know such trivial things :-D But who know....Anything can happen

+
[source,options="nowrap",subs="attributes+"]
----
Overwrite all contents of "/home/username/<OUTPUT_REPORT_DIRECTORY>" (anything already in the directory will be deleted)? [y,N]
----

However, if you specify the `--overwrite` argument, {ProductShortName} will proceed to delete and recreate the directory. See the description of this argument for more information.

[id="cli-source-argument_{context}"]
== Setting the source technology

> A space-delimited list of one or more source technologies, servers, platforms, or frameworks to migrate from. This argument, in conjunction with the `--target` argument, helps to determine which rulesets are used. Use the `--listSourceTechnologies` argument to list all available sources.

Reviewer 2: Our beloved IBM SG suggest to avoid anthropomorphism as much as we can -_- How about:

```suggestion
A space-delimited list of one or more source technologies, servers, platforms, or frameworks to migrate from. You can use this argument, in conjunction with the `--target` argument, to determine which rulesets are used. Use the `--listSourceTechnologies` argument to list all available sources.
```

.Usage
[source,options="nowrap",subs="attributes+"]
----
--source <SOURCE_1> <SOURCE_2>
----

> The `--source` argument now provides version support, which follows the link:http://maven.apache.org/enforcer/enforcer-rules/versionRanges.html[Maven version range syntax]. This instructs {ProductShortName} to only run the rulesets matching the specified versions. For example, `--source eap:5`.

Reviewer 2: ```suggestion
The `--source` argument now provides version support, which follows the link:http://maven.apache.org/enforcer/enforcer-rules/versionRanges.html[Maven version range syntax]. This instructs {ProductShortName} to only run the rulesets matching the specified versions, for example, `--source eap:5`.
```

[WARNING]
====
When migrating to JBoss EAP, be sure to specify the version, for example, `eap:6`. Specifying only `eap` will run rulesets for all versions of JBoss EAP, including those not relevant to your migration path.

See link:{ProductDocIntroToMTAGuideURL}/index#migration_paths_getting-started-guide[Supported migration paths] in _{IntroToMTABookName}_ for the appropriate JBoss EAP version.
====

[id="cli-target-argument_{context}"]
== Setting the target technology

> A space-delimited list of one or more target technologies, servers, platforms, or frameworks to migrate to. This argument, in conjunction with the `--source` argument, helps to determine which rulesets are used. If you do not specify this option, you are prompted to select a target. Use the `--listTargetTechnologies` argument to list all available targets.

Reviewer 2: ```suggestion
A space-delimited list of one or more target technologies, servers, platforms, or frameworks to migrate to. You can use this argument, in conjunction with the `--source` argument, to determine which rulesets are used. If you do not specify this option, you are prompted to select a target. Use the `--listTargetTechnologies` argument to list all available targets.
```

----

The `--target` argument now provides version support, which follows the link:http://maven.apache.org/enforcer/enforcer-rules/versionRanges.html[Maven version range syntax]. This instructs {ProductShortName} to only run the rulesets matching the specified versions. For example, `--target eap7`.

[WARNING]
====
When migrating to JBoss EAP, be sure to specify the version in the target, for example, `eap6`. Specifying only `eap` will run rulesets for all versions of JBoss EAP, including those not relevant to your migration path.

See link:{ProductDocIntroToMTAGuideURL}/index#migration_paths_getting-started-guide[Supported migration paths] in _{IntroToMTABookName}_ for the appropriate JBoss EAP version.
====

////

> [id="cli-packages-argument_{context}"]

Author: @istein1 -  section on *Packages* removed here 
This only leaves three references of `--packages` to resolve for [MTA-4220](https://issues.redhat.com/browse/MTA-4220)

----

> The `--target` argument now provides version support, which follows the link:http://maven.apache.org/enforcer/enforcer-rules/versionRanges.html[Maven version range syntax]. This instructs {ProductShortName} to only run the rulesets matching the specified versions. For example, `--target eap:7`.

Reviewer 2: ```suggestion
The `--target` argument now provides version support, which follows the link:http://maven.apache.org/enforcer/enforcer-rules/versionRanges.html[Maven version range syntax]. This instructs {ProductShortName} to only run the rulesets matching the specified versions, for example, `--target eap:7`.
```
